src/services/checkS3BucketData.ts

```diff
@@ -14,43 +15,58 @@ export const checkS3BucketData = async (
     dataRequestParams.dateTo
   )
 
-  //TODO: add handling for when there is no data available for requested dates
-  const requestedAuditBucketObjects = await Promise.all(
-    prefixes.map(
-      async (prefix) =>
-        await listS3Objects({
-          Bucket: getEnv('AUDIT_BUCKET_NAME'),
-          Prefix: prefix
-        })
-    )
-  ).then((objects: string[][]) => objects.flat())
+  const requestedAuditBucketObjects = await retrieveS3ObjectsForPrefixes(
+    prefixes,
+    getEnv('AUDIT_BUCKET_NAME')
+  )
 
-  const existingAnalysisBucketObjects = await Promise.all(
-    prefixes.map(
-      async (prefix) =>
-        await listS3Objects({
-          Bucket: getEnv('ANALYSIS_BUCKET_NAME'),
-          Prefix: prefix
-        })
-    )
-  ).then((objects: string[][]) => objects.flat())
+  const existingAnalysisBucketObjects = await retrieveS3ObjectsForPrefixes(
+    prefixes,
+    getEnv('ANALYSIS_BUCKET_NAME')
+  )
 
   console.log(
     'Objects present in analysis bucket:',
     existingAnalysisBucketObjects
   )
 
+  console.log('Objects present in auditBucket', requestedAuditBucketObjects)
   const objectsToCopy = requestedAuditBucketObjects.filter(
-    (object) => !existingAnalysisBucketObjects.includes(object)
+    (object) =>
+      !existingAnalysisBucketObjects.map((o) => o.Key).includes(object.Key)
   )
 
-  console.log('Objects to copy:', objectsToCopy)
+  const standardTierLocationsToCopy = objectsToCopy
+    .filter((o) => o.StorageClass === 'STANDARD')
+    .map((o) => o.Key as string)
```

Comment:
Looks like Key and StorageClass could be undefined: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/_object.html

The filter should be fine but is it worth checking we aren't adding undefined values into the final lists

Reply:
Yeah, I was confused as to how you could have an item in a bucket without a key, but I guess for safety we could filter out anything that doesn't have either of those properties set.

Reply:
Have sorted this now, added some filtering for any weird objects like this in our buckets